circle.yml

```diff
@@ -0,0 +1,14 @@
+machine:
+  pre:
+    - wget "http://ftp.debian.org/debian/pool/main/s/shellcheck/shellcheck_0.4.4-4_amd64.deb"
+    - sudo dpkg -i "shellcheck_0.4.4-4_amd64.deb"
+    - export PATH="/usr/local/bin:$PATH"
+    - git clone https://github.com/sstephenson/bats.git /tmp/bats && pushd /tmp/bats && sudo ./install.sh /usr/local && popd
+
+test:
+  override:
+    - curl -s https://raw.githubusercontent.com/18F/laptop/master/seekrets-install | bash -
```

Comment:
Sounds good. I ran into several issues while exploring this:

1) We can't run the script on Linux in Travis because it comes with git < 2.9.1, and I couldn't find out if it's possible to upgrade git

2) So then I tried Circle CI, which does have git > 2.9.1, but for some reason, the bats test is failing for some of the tests only. Any idea why?

Here is the report:

```
bats test/seekrets.bats
1..19
ok 1 no args gives usage instructions
ok 2 option --version prints version number
ok 3 config command with no options shows config
ok 4 rules command with no options gives a listing of rules
ok 5 git-seekrets does find aws secrets in test repo
ok 6 git-seekrets does find aws accounts in test repo
ok 7 git-seekrets does find aws access keys in test repo
not ok 8 git-seekrets does not find newrelic keys as aws keys in test repo
# (in test file test/seekrets.bats, line 66)
#   `[ $status -eq 0 ]' failed
# Initialized empty Git repository in /tmp/gittest.PGPuwo/.git/
# Enabling all rules.
# List of rules:
# 	[x] aws.secret_access_key
# 	[x] aws.access_key_id
# 	[x] aws.account_id
# 	[x] mandrill.username
# 	[x] mandrill.password
# 	[x] mandrill.api_key
# 	[x] newrelic.license_key
# Disabling all rules.
# List of rules:
# 	[ ] aws.secret_access_key
# 	[ ] aws.access_key_id
# 	[ ] aws.account_id
# 	[ ] mandrill.api_key
# 	[ ] mandrill.username
# 	[ ] mandrill.password
# 	[ ] newrelic.license_key
# List of rules:
# 	[ ] aws.secret_access_key
# 	[x] aws.access_key_id
# 	[ ] aws.account_id
# 	[ ] mandrill.username
# 	[ ] mandrill.password
# 	[ ] mandrill.api_key
# 	[ ] newrelic.license_key
# List of rules:
# 	[ ] aws.secret_access_key
# 	[x] aws.access_key_id
# 	[x] aws.account_id
# 	[ ] mandrill.username
# 	[ ] mandrill.password
# 	[ ] mandrill.api_key
# 	[ ] newrelic.license_key
# List of rules:
# 	[x] aws.secret_access_key
# 	[x] aws.access_key_id
# 	[x] aws.account_id
# 	[ ] mandrill.username
# 	[ ] mandrill.password
# 	[ ] mandrill.api_key
# 	[ ] newrelic.license_key
ok 9 git-seekrets does find newrelic secrets in test repo
not ok 10 git-seekrets does not find newrelic false positives in test repo
# (in test file test/seekrets.bats, line 77)
#   `[ $status -eq 0 ]' failed
# Initialized empty Git repository in /tmp/gittest.AI8qF8/.git/
# Enabling all rules.
# List of rules:
# 	[x] aws.account_id
# 	[x] aws.secret_access_key
# 	[x] aws.access_key_id
# 	[x] mandrill.username
# 	[x] mandrill.password
# 	[x] mandrill.api_key
# 	[x] newrelic.license_key
not ok 11 git-seekrets only matches newrelic secrets in test repo
# (in test file test/seekrets.bats, line 84)
#   `[ $(echo "$output" | grep -c 'Found Secrets: 1') -eq 1 ]' failed
# Initialized empty Git repository in /tmp/gittest.ipBXLZ/.git/
# Enabling all rules.
# List of rules:
# 	[x] aws.secret_access_key
# 	[x] aws.access_key_id
# 	[x] aws.account_id
# 	[x] mandrill.username
# 	[x] mandrill.password
# 	[x] mandrill.api_key
# 	[x] newrelic.license_key
ok 12 git-seekrets does find mandrill keys in test repo
not ok 13 git-seekrets does not find mandrill key false positives in test repo
# (in test file test/seekrets.bats, line 94)
#   `[ $status -eq 0 ]' failed
# Initialized empty Git repository in /tmp/gittest.U8Etba/.git/
# Enabling all rules.
# List of rules:
# 	[x] aws.secret_access_key
# 	[x] aws.access_key_id
# 	[x] aws.account_id
# 	[x] mandrill.username
# 	[x] mandrill.password
# 	[x] mandrill.api_key
# 	[x] newrelic.license_key
ok 14 git-seekrets does find mandrill passwords in test repo
not ok 15 git-seekrets does not find mandrill password false positives in test repo
# (in test file test/seekrets.bats, line 105)
#   `[ $status -eq 0 ]' failed
# Initialized empty Git repository in /tmp/gittest.qDKRhV/.git/
# Enabling all rules.
# List of rules:
# 	[x] aws.access_key_id
# 	[x] aws.account_id
# 	[x] aws.secret_access_key
# 	[x] mandrill.username
# 	[x] mandrill.password
# 	[x] mandrill.api_key
# 	[x] newrelic.license_key
ok 16 git-seekrets does find mandrill usernames in test repo
not ok 17 git-seekrets does not find secrets in test repo
# (in test file test/seekrets.bats, line 116)
#   `[ $status -eq 0 ]' failed
# Initialized empty Git repository in /tmp/gittest.EX1sUn/.git/
# Enabling all rules.
# List of rules:
# 	[x] aws.secret_access_key
# 	[x] aws.access_key_id
# 	[x] aws.account_id
# 	[x] mandrill.username
# 	[x] mandrill.password
# 	[x] mandrill.api_key
# 	[x] newrelic.license_key
ok 18 git-seekrets can disable all rulesets
ok 19 git-seekrets can enable all rulesets

bats test/seekrets.bats returned exit code 1
```

If we can't resolve why some of the bats tests are failing on Ubuntu, I'm going to look into how we can use an OS X image on Circle CI and compare the speed to Travis. I can't stand waiting this long for the tests to run.